### PR TITLE
Feature Request: Adding Mish and Swish Activation function

### PR DESCRIPTION
**Pull request by digantamisra98**
Mish is a new novel activation function proposed in this [paper](https://arxiv.org/abs/1908.08681).
It has shown promising results so far and has been adopted in several packages including:

- [TensorFlow-Addons](https://github.com/tensorflow/addons/tree/master/tensorflow_addons/activations)
- [SpaCy (Tok2Vec Layer)](https://github.com/explosion/spaCy)
- [Thinc - SpaCy's official NLP based ML library](https://github.com/explosion/thinc/releases/tag/v7.3.0)
- [Echo AI](https://github.com/digantamisra98/Echo)
- [CNTKX - Extension of Microsoft's CNTK](https://github.com/delzac/cntkx)
- [FastAI-Dev](https://github.com/fastai/fastai_dev/blob/0f613ba3205990c83de9dba0c8798a9eec5452ce/dev/local/layers.py#L441)
- [Darknet](https://github.com/AlexeyAB/darknet/commit/bf8ea4183dc265ac17f7c9d939dc815269f0a213)
- [Yolov3](https://github.com/ultralytics/yolov3/commit/444a9f7099d4ff1aef12783704e3df9a8c3aa4b3)
- [BeeDNN - Library in C++](https://github.com/edeforas/BeeDNN)
- [Gen-EfficientNet-PyTorch](https://github.com/rwightman/gen-efficientnet-pytorch)
- [dnet](https://github.com/umangjpatel/dnet/blob/master/nn/activations.py)

All benchmarks, analysis and links to official package implementations can be found in this [repository](https://github.com/digantamisra98/Mish) 

It would be nice to have Mish as an option within the activation function group.

This is the comparison of Mish with other conventional activation functions in a SEResNet-50 for CIFAR-10: 
se50_1

Swish activation was proposed by Google Brain in this paper - https://arxiv.org/abs/1710.05941

### CODE REVIEW

File: allennlp/nn/activations.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def __call__(self, tensor: torch.Tensor) -> torch.Tensor:
 # Activation.by_name('relu')()
 Registrable._registry[Activation] = {
     "linear": lambda: lambda x: x,  # type: ignore
+    "mish": lambda: lambda x: x * torch.tanh(torch.nn.functional.softplus(x)),
```
Comment (DeNeutoy) [inline allennlp/nn/activations.py]: You will need to add this `# type: ignore` statement to these lambda functions for our mypy check.

Comment (digantamisra98) [reply allennlp/nn/activations.py]: Alright. Will make the changes.

Comment (digantamisra98) [reply allennlp/nn/activations.py]: @DeNeutoy need a bit of help with running flake8 and black tests. Whenever I try it locally, it says module import error that no module named allennlp is found although allennlp is installed and there is no name conflicts. I fixed the code as per your review though.

**Comment by digantamisra98** [comment]
@DeNeutoy need a bit of help with running flake8 and black tests. Whenever I try it locally, it says module import error that no module named allennlp is found although allennlp is installed and there is no name conflicts. I fixed the code as per your review though.

**Comment by DeNeutoy** [comment]
@digantamisra98 ah sorry you need to run `flake8 allennlp` and `black allennlp`, i.e run it on the `allennlp` directory which contains all the code.

**Comment by digantamisra98** [comment]
@DeNeutoy ah alright. On the road, will do that in the next week probably. I added the type ignore statement for mypy check though.

**Comment by schmmd** [comment]
@DeNeutoy do you approve this PR now that it's passing tests?

**Comment by digantamisra98** [comment]
@schmmd Thank you for the help. Had completely forgotten about this PR due to recent commitments. Apologies for any inconveniences caused.

**Pull request merged**